eth/blockwatch/block_watcher.go

```diff
@@ -159,6 +159,12 @@ func (w *Watcher) syncToLatestBlock() error {
 		return err
 	}
 
+	if lastSeenHeader == nil {
+		lastSeenHeader = &MiniHeader{
```

Comment:
Looks like this would result in `newestHeader.Number` calls to `pollNextBlock()` when `lastSeenHeader` is nil i.e. the number of blocks since genesis.

If `lastSeenHeader` is nil we would only want to call `pollNextBlock()` up to `newestHeader.Number - w.startBlockDepth` times (`pollNextBlock()` also starts from `w.startBlockDepth` if the latest header from the DB is nil).

Reply:
Fixed in 0e11f59